src/h3lib/lib/latLng.c

```diff
@@ -259,47 +259,69 @@ void _geoAzDistanceRads(const LatLng *p1, double az, double distance,
  * the future.
  */
 
-double H3_EXPORT(getHexagonAreaAvgKm2)(int res) {
+H3Error H3_EXPORT(getHexagonAreaAvgKm2)(int res, double *out) {
     static const double areas[] = {
         4.357449416078383e+06, 6.097884417941332e+05, 8.680178039899720e+04,
         1.239343465508816e+04, 1.770347654491307e+03, 2.529038581819449e+02,
         3.612906216441245e+01, 5.161293359717191e+00, 7.373275975944177e-01,
         1.053325134272067e-01, 1.504750190766435e-02, 2.149643129451879e-03,
         3.070918756316060e-04, 4.387026794728296e-05, 6.267181135324313e-06,
         8.953115907605790e-07};
-    return areas[res];
+    if (res < 0 || res > MAX_H3_RES) {
+        return E_RES_DOMAIN;
+    }
+    *out = areas[res];
+    return E_SUCCESS;
 }
 
-double H3_EXPORT(getHexagonAreaAvgM2)(int res) {
+H3Error H3_EXPORT(getHexagonAreaAvgM2)(int res, double *out) {
```

Comment:
This function - amongst possibly others - doesn't seem to have been renamed in the docs. We should check that functions have been renamed & all functions are listed in the docs with up to date signatures.